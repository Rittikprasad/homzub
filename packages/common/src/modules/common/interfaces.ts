import { IRedirectionDetails } from '@homzhub/mobile/src/services/LinkingService';
import { IRedirectionDetailsWeb } from '@homzhub/web/src/services/NavigationService';
import { ICountry } from '@homzhub/common/src/domain/models/Country';
import { IfscDetail } from '@homzhub/common/src/domain/models/IfscDetail';
import { IMessages, Messages } from '@homzhub/common/src/domain/models/Message';
import { GroupMessage } from '@homzhub/common/src/domain/models/GroupMessage';
import { IPillar } from '@homzhub/common/src/domain/models/Pillar';

export interface ICommonState {
  countries: ICountry[];
  deviceCountry: string;
  messages: IMessages | null;
  attachment: string;
  redirectionDetails: IRedirectionDetails | IRedirectionDetailsWeb;
  currentChatDetail: IChatPayload | null;
  groupMessages: GroupMessage[] | null;
  pillars: IPillar[];
  reviewReferData: IReviewRefer | null;
  ifscDetail: IfscDetail;
  loaders: {
    groupMessages: boolean;
    messages: boolean;
    whileGetCountries: boolean;
    pillars: boolean;
  };
}

export interface IMessageSuccess {
  response: Messages;
  isNew?: boolean;
}

export interface IChatPayload {
  groupName: string;
  groupId: number;
}

export interface IReviewRefer {
  message?: string;
  isReview?: boolean;
  isSheetVisible?: boolean;
}
