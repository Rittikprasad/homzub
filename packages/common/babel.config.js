module.exports = {
  
  presets: ['@babel/preset-env', '@babel/preset-react', '@babel/preset-typescript'],
  plugins: [['@babel/plugin-proposal-decorators', { legacy: true }],["@babel/plugin-proposal-class-properties", { "loose": true }],["@babel/plugin-transform-flow-strip-types"]],
  env: {
    test: {
      plugins: [['@babel/plugin-proposal-decorators', { legacy: true }],["@babel/plugin-proposal-class-properties", { "loose": true }]],
      presets: ['@babel/preset-env', '@babel/preset-react', '@babel/preset-typescript'],
      // ignore: [/node_modules/],
    },
  },
};
