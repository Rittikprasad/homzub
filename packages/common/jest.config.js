/* eslint-disable @typescript-eslint/no-var-requires */
const baseJestConfig = require('../../jest.config');

module.exports = {
  ...baseJestConfig,
};

// jest.config.js
module.exports = {
  moduleNameMapper: {
    "\\.svg": "<rootDir>/__mocks__/svgMock.js"
  }
};