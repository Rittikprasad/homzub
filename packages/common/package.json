{
  "name": "@homzhub/common",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "test": "jest --coverage --env=jsdom",
    "lint": "eslint 'src/**/*.{js,ts,tsx}'",
    "lint:fix": "eslint 'src/**/*.{js,ts,tsx}' --fix",
    "format": "prettier --write './**/*.{ts,tsx,js,json}'",
    "sonar": "ts-node sonar-project.ts"
  },
  "jestSonar": {
    "reportPath": "coverage",
    "reportFile": "test-reporter.xml",
    "indent": 4
  }
}
