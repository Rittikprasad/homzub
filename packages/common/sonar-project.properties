sonar.projectKey=Homzhub-FE-Common
sonar.projectName=Homzhub-FE-Common
sonar.projectVersion=1.0
# Comma-separated paths to directories with sources (required)
sonar.sources=packages/common/src
# Exclude files
sonar.exclusions=packages/common/node_modules/**,packages/common/src/mocks/*
# Language
sonar.language=ts
# Encoding of the source files
sonar.sourceEncoding=UTF-8
sonar.coverage.exclusions=packages/common/src/mocks/*,packages/common/src/constants/*,packages/common/src/assets/*,packages/common/node_modules/*,packages/common/coverage/lcov-report/*,packages/common/src/domain/models/*,packages/common/src/styles/*
sonar.typescript.lcov.reportPaths=packages/common/coverage/lcov.info
sonar.testExecutionReportPaths=packages/common/coverage/test-reporter.xml
sonar.test.inclusions=packages/common/src/**/*.test.ts,packages/common/src/**/*.test.tsx
