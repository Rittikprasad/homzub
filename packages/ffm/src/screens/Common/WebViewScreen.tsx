import React, { useCallback, useEffect } from 'react';
import { WebView } from 'react-native-webview';
import { StyleSheet, View, Share } from 'react-native';
import { useTranslation } from 'react-i18next';
import { icons } from '@homzhub/common/src/assets/icon';
import { Header } from '@homzhub/mobile/src/components';
import { AuthStackParamList } from '@homzhub/ffm/src/navigation/AuthStack';
import { NavigationScreenProps, ScreenKeys } from '@homzhub/ffm/src/navigation/interfaces';

type Props = NavigationScreenProps<AuthStackParamList, ScreenKeys.WebViewScreen>;

export const WebViewScreen = (props: Props): React.ReactElement => {
  const {
    navigation,
    route: {
      params: { url },
    },
  } = props;
  const { t } = useTranslation();

  const handleBackPress = useCallback((): void => {
    navigation.goBack();
  }, [navigation]);

  const onShare = useCallback(async (): Promise<void> => {
    await Share.share({
      message: `${t('assetMore:shareTrend')} ${url}`,
    });
  }, [url, t]);

  return (
    <View style={styles.container}>
      <Header icon={icons.leftArrow} onIconPress={handleBackPress} />
      <WebView source={{ uri: url }} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
