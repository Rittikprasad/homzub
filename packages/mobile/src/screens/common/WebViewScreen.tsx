import React, { useCallback, useEffect } from 'react';
import { WebView } from 'react-native-webview';
import { StyleSheet, View, Share } from 'react-native';
import { useTranslation } from 'react-i18next';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { ErrorUtils } from '@homzhub/common/src/utils/ErrorUtils';
import { CommonRepository } from '@homzhub/common/src/domain/repositories/CommonRepository';
import { icons } from '@homzhub/common/src/assets/icon';
import { Header } from '@homzhub/mobile/src/components';
import { Loader } from '@homzhub/common/src/components/atoms/Loader';
import { AuthStackParamList } from '@homzhub/mobile/src/navigation/AuthStack';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';

type Props = NavigationScreenProps<AuthStackParamList, ScreensKeys.WebViewScreen>;

export const WebViewScreen = (props: Props): React.ReactElement => {
  const {
    navigation,
    route: {
      params: { url, trendId },
    },
  } = props;
  const { t } = useTranslation();

  useEffect(() => {
    if (!trendId) {
      return;
    }
    try {
      CommonRepository.updateMarketTrends(trendId).then();
    } catch (err) {
      AlertHelper.error({ message: ErrorUtils.getErrorMessage(err.details) });
    }
  }, []);

  const handleBackPress = useCallback((): void => {
    navigation.goBack();
  }, [navigation]);

  const onShare = useCallback(async (): Promise<void> => {
    await Share.share({
      message: `${t('assetMore:shareTrend')} ${url}`,
    });
  }, [url, t]);

  const ActivityIndicatorLoadingView = useCallback((): React.ReactElement => <Loader visible />, []);

  return (
    <View style={styles.container}>
      <Header
        icon={icons.leftArrow}
        onIconPress={handleBackPress}
        iconRight={trendId ? icons.shareExternal : undefined}
        onIconRightPress={trendId ? onShare : undefined}
      />
      <WebView source={{ uri: url }} renderLoading={ActivityIndicatorLoadingView} startInLoadingState />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
