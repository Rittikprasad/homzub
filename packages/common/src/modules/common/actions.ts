import { ObjectMapper } from '@homzhub/common/src/utils/ObjectMapper';
import { IRedirectionDetails } from '@homzhub/mobile/src/services/LinkingService';
import { IRedirectionDetailsWeb } from '@homzhub/web/src/services/NavigationService';
import { Country, ICountry } from '@homzhub/common/src/domain/models/Country';
import { GroupMessage } from '@homzhub/common/src/domain/models/GroupMessage';
import { IfscDetail } from '@homzhub/common/src/domain/models/IfscDetail';
import { Messages } from '@homzhub/common/src/domain/models/Message';
import { IPillar, Pillar, PillarTypes } from '@homzhub/common/src/domain/models/Pillar';
import { IGetMessageParam } from '@homzhub/common/src/domain/repositories/interfaces';
import { IChatPayload, IMessageSuccess, IReviewRefer } from '@homzhub/common/src/modules/common/interfaces';
import { IFluxStandardAction } from '@homzhub/common/src/modules/interfaces';

const actionTypePrefix = 'Common/';
export const CommonActionTypes = {
  GET: {
    COUNTRIES: `${actionTypePrefix}COUNTRIES`,
    COUNTRIES_SUCCESS: `${actionTypePrefix}COUNTRIES_SUCCESS`,
    COUNTRIES_FALIURE: `${actionTypePrefix}COUNTRIES_FALIURE`,
    MESSAGES: `${actionTypePrefix}MESSAGES`,
    MESSAGES_SUCCESS: `${actionTypePrefix}MESSAGES_SUCCESS`,
    GROUP_MESSAGES: `${actionTypePrefix}GROUP_MESSAGES`,
    GROUP_MESSAGES_SUCCESS: `${actionTypePrefix}GROUP_MESSAGES_SUCCESS`,
    PILLARS: `${actionTypePrefix}PILLARS`,
    PILLARS_SUCCESS: `${actionTypePrefix}PILLARS_SUCCESS`,
    PILLARS_FAILURE: `${actionTypePrefix}PILLARS_FAILURE`,
  },
  SET: {
    DEVICE_COUNTRY: `${actionTypePrefix}DEVICE_COUNTRY`,
    REDIRECTION_DETAILS: `${actionTypePrefix}REDIRECTION_DETAILS`,
    MESSAGE_ATTACHMENT: `${actionTypePrefix}MESSAGE_ATTACHMENT`,
    CURRENT_CHAT: `${actionTypePrefix}CURRENT_CHAT`,
    REVIEW_REFER_DATA: `${actionTypePrefix}REVIEW_REFER_DATA`,
    IFSC_DETAIL: `${actionTypePrefix}IFSC_DETAIL`,
  },
  CLEAR_MESSAGES: `${actionTypePrefix}CLEAR_MESSAGES`,
  CLEAR_ATTACHMENT: `${actionTypePrefix}CLEAR_ATTACHMENT`,
  CLEAR_CHAT_DETAIL: `${actionTypePrefix}CLEAR_CHAT_DETAIL`,
  CLEAR_IFSC_DETAIL: `${actionTypePrefix}CLEAR_IFSC_DETAIL`,
};

const getCountries = (): IFluxStandardAction => ({
  type: CommonActionTypes.GET.COUNTRIES,
});

const getCountriesSuccess = (payload: Country[]): IFluxStandardAction<ICountry[]> => ({
  type: CommonActionTypes.GET.COUNTRIES_SUCCESS,
  payload: ObjectMapper.serializeArray(payload),
});
const getCountriesFaliure = (error: string): IFluxStandardAction => ({
  type: CommonActionTypes.GET.COUNTRIES_FALIURE,
  error,
});
const setDeviceCountry = (countryCode: string): IFluxStandardAction<string> => ({
  type: CommonActionTypes.SET.DEVICE_COUNTRY,
  payload: countryCode,
});

const setRedirectionDetails = (
  payload: IRedirectionDetails | IRedirectionDetailsWeb
): IFluxStandardAction<IRedirectionDetails | IRedirectionDetailsWeb> => ({
  type: CommonActionTypes.SET.REDIRECTION_DETAILS,
  payload,
});

const getMessages = (payload: IGetMessageParam): IFluxStandardAction<IGetMessageParam> => ({
  type: CommonActionTypes.GET.MESSAGES,
  payload,
});

const getMessagesSuccess = (payload: IMessageSuccess): IFluxStandardAction<IMessageSuccess> => ({
  type: CommonActionTypes.GET.MESSAGES_SUCCESS,
  payload,
});

const setAttachment = (payload: string): IFluxStandardAction<string> => ({
  type: CommonActionTypes.SET.MESSAGE_ATTACHMENT,
  payload,
});

const clearMessages = (): IFluxStandardAction => ({
  type: CommonActionTypes.CLEAR_MESSAGES,
});

const clearAttachment = (): IFluxStandardAction => ({
  type: CommonActionTypes.CLEAR_ATTACHMENT,
});

// TODO: (Shivam: 23/2/21: add types)
const getGroupMessage = (): IFluxStandardAction => ({
  type: CommonActionTypes.GET.GROUP_MESSAGES,
});

const getGroupMessageSuccess = (groupMessages: GroupMessage[]): IFluxStandardAction<GroupMessage[]> => ({
  type: CommonActionTypes.GET.GROUP_MESSAGES_SUCCESS,
  payload: ObjectMapper.serializeArray(groupMessages),
});

const setCurrentChatDetail = (payload: IChatPayload): IFluxStandardAction<IChatPayload> => ({
  type: CommonActionTypes.SET.CURRENT_CHAT,
  payload,
});

const clearChatDetail = (): IFluxStandardAction => ({
  type: CommonActionTypes.CLEAR_CHAT_DETAIL,
});

const getPillars = (payload: PillarTypes): IFluxStandardAction<PillarTypes> => ({
  type: CommonActionTypes.GET.PILLARS,
  payload,
});

const getPillarsSuccess = (payload: Pillar[]): IFluxStandardAction<IPillar[]> => ({
  type: CommonActionTypes.GET.PILLARS_SUCCESS,
  payload: ObjectMapper.serializeArray(payload),
});

const getPillarsFailure = (): IFluxStandardAction => ({
  type: CommonActionTypes.GET.PILLARS_FAILURE,
});

const setReviewReferData = (payload: IReviewRefer): IFluxStandardAction<IReviewRefer> => ({
  type: CommonActionTypes.SET.REVIEW_REFER_DATA,
  payload,
});

const setIfscDetail = (payload: IfscDetail): IFluxStandardAction<IfscDetail> => ({
  type: CommonActionTypes.SET.IFSC_DETAIL,
  payload,
});

const clearIfscDetail = (): IFluxStandardAction => ({
  type: CommonActionTypes.CLEAR_IFSC_DETAIL,
});

export type CommonActionPayloadTypes =
  | ICountry[]
  | IRedirectionDetails
  | IGetMessageParam
  | IMessageSuccess
  | Messages
  | GroupMessage[]
  | IChatPayload
  | PillarTypes
  | IPillar[]
  | IReviewRefer
  | string
  | IfscDetail;

export const CommonActions = {
  getCountries,
  getCountriesSuccess,
  getCountriesFaliure,
  setDeviceCountry,
  setRedirectionDetails,
  getMessages,
  getMessagesSuccess,
  clearMessages,
  setAttachment,
  clearAttachment,
  getGroupMessage,
  getGroupMessageSuccess,
  setCurrentChatDetail,
  clearChatDetail,
  getPillars,
  getPillarsSuccess,
  getPillarsFailure,
  setReviewReferData,
  setIfscDetail,
  clearIfscDetail,
};
