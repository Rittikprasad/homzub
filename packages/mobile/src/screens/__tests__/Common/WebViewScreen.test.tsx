import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { WebViewScreen } from '@homzhub/mobile/src/screens/common/WebViewScreen';

const mock = jest.fn();

describe('Web View Screen', () => {
  let component: ShallowWrapper;
  let props: any;

  beforeEach(() => {
    props = {
      navigation: {
        goBack: mock,
      },
      route: {
        params: {
          url: '',
        },
      },
    };
    component = shallow(<WebViewScreen {...props} />);
  });

  it('should render snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
