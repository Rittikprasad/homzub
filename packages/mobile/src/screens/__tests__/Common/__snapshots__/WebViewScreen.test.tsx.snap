// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Web View Screen should render snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Memo(Header)
    icon="left-arrow"
    onIconPress={[Function]}
  />
  <WebView
    cacheEnabled={true}
    javaScriptEnabled={true}
    originWhitelist={
      Array [
        "http://*",
        "https://*",
      ]
    }
    renderLoading={[Function]}
    source={
      Object {
        "uri": "",
      }
    }
    startInLoadingState={true}
    useSharedProcessPool={true}
  />
</View>
`;
